internal/k6runner/k6runner.go

```diff
@@ -307,8 +313,18 @@ NEXT_RECORD:
 type HttpRunner struct {
 	url    string
 	logger *zerolog.Logger
+	// backoff sets the minimum amount of time to wait before retrying a request. nth attempt waits n times this value,
+	// plus some jitter.
+	backoff time.Duration
+	// graceTime tells the HttpRunner how much time to add do the script timeout to form the request timeout.
+	graceTime time.Duration
 }
 
+const (
+	defaultBackoff   = 10 * time.Second
```

Comment:
This is a bit of "wet finger in the wind". The final intent of this mechanism is to retry infrastructure faults, such as runners being updated, or half of them being unavailable because a node fell off the network. My thoughts were that these kind of failures are more of the "will be fixed given some time" type, rather than "will be fixed if I retry enough times. In fact, each retry adds an item to the queue, which makes the system slower (thus the need of a backoff).

_Intuitively_ I think something that fails on `t.Now()` will likely also fail in `t.Now() + 3*time.Second`, because most of these failures will take several seconds to reconcile.

> I was thinking how we could have some observability to understand whether this is being effective or not, but that seems tricky to make it "understandable", and it depends on the user configuration for the check also...

This is not a bad idea I think. An historgram reporting retries per status code could be useful I think. WDYT? I think that should be possible to implement in a follow-up PR and might help us fine-tune the initial number and scaling.

Reply:
Why would it be per status code? I'm not getting that. My initial thought was that what I want to know is if the retries are successful in order to complete the execution, so maybe a histogram for retries and "status" where this one is OK/KO whether the request was eventually completed successfully?